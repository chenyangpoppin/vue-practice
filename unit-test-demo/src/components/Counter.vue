<template>
  <div>
    <div>Count: {{ count }}</div>
    <button @click="() => increseCount()">Count++</button>
  </div>
</template>

<script>
export default {
  name: 'Counter',
  data() {
    return {
      count: 0
    };
  },
  methods: {
    increseCount: function() {
      this.count++;
    }
  }
}
</script>