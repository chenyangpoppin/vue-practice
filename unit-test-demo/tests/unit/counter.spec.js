import { mount } from "@vue/test-utils";
import Counter from "@/components/Counter.vue";
import sinon from "sinon";

describe("Counter.vue", () => {
  const change = sinon.spy();

  const wrapper = mount(Counter, {
    listeners: {
      change
    }
  });

  it("[UT Counter]: Renders counter component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  /**
   * Test count value after clicking the increase button
   */
  it("[UT Counter]: Event handler of count++", () => {
    const button = wrapper.find("button");

    // Click the button for the very first time
    button.trigger("click");
    expect(wrapper.vm.count).toBe(1);
    expect(change.called).toBe(true);

    // Click the button for the second time
    button.trigger("click");
    expect(change.callCount).toBe(2);
  });
});
