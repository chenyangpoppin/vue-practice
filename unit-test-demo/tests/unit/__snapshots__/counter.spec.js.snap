// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue [UT Counter]: Renders counter component 1`] = `
<div>
  <div>Count: 0</div> <button>Count++</button>
</div>
`;

exports[`Counter.vue renders counter component 1`] = `
<div>
  <div>Count: 0</div> <button>Count++</button>
</div>
`;
